apps/tmbitwarden.sh

```diff
@@ -78,22 +92,145 @@ cd /root
 curl_to_dir "https://raw.githubusercontent.com/bitwarden/core/master/scripts" "bitwarden.sh" "/root"
 chmod +x /root/bitwarden.sh
 check_command ./bitwarden.sh install
-sed -i "s|http_port.*|http_port: 8080|g" /root/bwdata/config.yml
-sed -i "s|https_port.*|https_port: 8443|g" /root/bwdata/config.yml
+
+# Check if all ssl settings were entered correctly
+if grep ^url /root/bwdata/config.yml | grep -q https || grep ^url /root/bwdata/config.yml | grep -q localhost
+then
+    message "It seems like you have entered some wrong settings. We will remove bitwarden now again so that you can start over again."
+    check_command docker rm bitwarden-nginx bitwarden-admin bitwarden-events bitwarden-attachments \
+    bitwarden-identity bitwarden-api bitwarden-web bitwarden-icons bitwarden-notifications bitwarden-mssql
+    rm -rf /root/bwdata
+fi
+
+# Continue with the installation
+sed -i "s|http_port.*|http_port: 5178|g" /root/bwdata/config.yml
+sed -i "s|https_port.*|https_port: 5179|g" /root/bwdata/config.yml
+# Get Subdomain from config.yml and change it to https
+SUBDOMAIN=$(grep ^url /root/bwdata/config.yml)
+SUBDOMAIN=${SUBDOMAIN##*url: http://}
+sed -i "s|^url: .*|url: https://$SUBDOMAIN|g" /root/bwdata/config.yml
+sed -i 's|http://|https://|g' /root/bwdata/env/global.override.env
 check_command ./bitwarden.sh rebuild
 check_command ./bitwarden.sh start
-if check_command ./bitwarden.sh updatedb
+check_command ./bitwarden.sh updatedb
```

Comment:
BTW: if you really don't like that we already start bitwarden here, although the ssl certificate is not existing, we could also do that later. 

But maybe it isn't that important since the proxy to the correct port happens later when proxying it 😅

Reply:
What I mean is thtat we could move these three lines to the complete end of the script if we wanted ;)

Reply:
Idk, but Bitwarden doesn't have to start until the end anyway, but it might be a good idea to start if before we begin with the proxy so that we don't clutter the system when not needed.

Also, now when thinking about it, if the Bitwarden installation fails, we actually should remove `/bwdata` so that the script can be run again, since no passwords are lost anyway. Right?

Reply:
Aah, just saw that you did that now. Great!

Reply:
yup